.github/workflows/build.yml

```diff
@@ -0,0 +1,353 @@
+
+      - name: Install Yarn
+        run: |
+          npm install -g yarn
+      
+      - name: UI Build
+        run: |
+          cd ui
+          yarn install --network-timeout 300000
+          yarn --verbose run build
+          cd ..
+      
+      - name: Build prerelease
+        run: make prerelease
+      
+      - name: Install Linux build utilties
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y software-properties-common
+          sudo dpkg --add-architecture i386
+          sudo apt-get update
+          sudo apt-get install -y \
+            libc6-dev-i386 \
+            libpcre3-dev \
+            linux-libc-dev:i386 
+          sudo apt-get install -y \
+            binutils-aarch64-linux-gnu \
+            binutils-arm-linux-gnueabihf \
+            gcc-aarch64-linux-gnu \
+            gcc-arm-linux-gnueabihf \
+            gcc-multilib-arm-linux-gnueabihf
+          sudo apt-get upgrade -y
+
+      - name: Set gcc
+        run: |
+          if [ "${{ matrix.goarch }}" == "arm" ]; then
+            echo "CC=arm-linux-gnueabihf-gcc" >> $GITHUB_ENV
+          elif [ "${{ matrix.goarch }}" == "arm64" ]; then
+            echo "CC=aarch64-linux-gnu-gcc" >> $GITHUB_ENV
+          fi
+
+      - name: Build
+        env:
+          GOOS: ${{ matrix.goos }}
+          CC: ${{ env.CC }}
+          GOARCH: ${{ matrix.goarch }}
+          LD_FLAGS: ${{ needs.set-ld-flags.outputs.ldflags }}
+          CGO_ENABLED: 1
+        run: |
+          mkdir -p dist out
+          go build -o dist/ \
+          -ldflags "${{ env.LD_FLAGS }}" \
+          -tags "${{ env.GO_TAGS }}"
+          zip -r -j out/${{ env.PKG_NAME }}_${{ needs.get-product-version.outputs.product-version }}_${{ matrix.goos }}_${{ matrix.goarch }}.zip dist/
+      - uses: actions/upload-artifact@v2
+        with:
+          name: ${{ env.PKG_NAME }}_${{ needs.get-product-version.outputs.product-version }}_${{ matrix.goos }}_${{ matrix.goarch }}.zip
+          path: out/${{ env.PKG_NAME }}_${{ needs.get-product-version.outputs.product-version }}_${{ matrix.goos }}_${{ matrix.goarch }}.zip
+
+      - name: Package
+        if: ${{ matrix.goos == 'linux' }}
+        uses: hashicorp/actions-packaging-linux@v1
+        with:
+          name: ${{ env.PKG_NAME }}
+          description: "Nomad is an easy-to-use, flexible, and performant workload orchestrator that can deploy a mix of microservice, batch, containerized, and non-containerized applications."
+          arch: ${{ matrix.goarch }}
+          version: ${{ needs.get-product-version.outputs.product-version }}
+          maintainer: "HashiCorp"
+          homepage: "https://github.com/hashicorp/nomad"
+          license: "MPL-2.0"
+          binary: "dist/${{ env.PKG_NAME }}"
+          deb_depends: "openssl"
+          rpm_depends: "openssl"
+          config_dir: ".release/linux/package/"
+          preinstall: ".release/linux/preinst"
+          postremove: ".release/linux/postrm"
```

Comment:
We've been thinking about how to improve our Linux packaging story, and one of the ideas was to possibly have two different packages, one for Nomad servers and one for clients.

If we decide to go down that route, would it be possible to add another step like this, but pointing to different directories?

(some quick background, Nomad client and servers have very distinct environment setup. For example, Nomad servers should run under its own user, while Clients run as `root`. They also usually don't run in the same machine, so having separate packages may help users install the base configuration that they need).

Reply:
This is a great question for @kpenfound as he is our Linux packaging SME.